src/components/typography/typography.tsx

```diff
@@ -42,124 +44,145 @@ export interface TypographyProps extends HTMLAttributes<HTMLElement> {
   hasCapitalisedText?: boolean
   hasUpperCasedText?: boolean
   intent?: Intent
+  tag?: TypeographyTag
+}
+
+export interface TaggedTypographyProps extends TypographyProps {
+  baseClass?: string
 }
 
-const propsToClass = ({
-  className,
-  hasGreyText,
-  hasItalicText,
-  hasBoldText,
-  hasRegularText,
-  hasMediumText,
-  hasNoMargin,
-  hasMargin,
-  hasSectionMargin,
-  hasCenteredText,
-  hasDisabledText,
-  hasCapitalisedText,
-  hasUpperCasedText,
-  intent,
-  children,
-  ...rest
-}: TypographyProps) => () => {
-  const intentClass = intent ? getIntentClassName(intent) : null
-
-  return {
-    className: cx(
-      className,
-      hasGreyText && elHasGreyText,
-      hasRegularText && elHasRegularText,
-      hasBoldText && elHasBoldText,
-      hasItalicText && elHasItalicText,
-      hasMediumText && elHasMediumText,
-      hasMargin && elHasMargin,
-      hasNoMargin && elHasNoMargin,
-      hasSectionMargin && elHasSectionMargin,
-      hasCenteredText && elHasCenteredText,
-      hasDisabledText && elHasDisabledText,
-      hasCapitalisedText && elHasCapitalisedText,
-      hasUpperCasedText && elHasUpperCasedText,
-      intentClass,
-    ),
+const propsToClass =
+  ({
+    baseClass,
+    className,
+    hasGreyText,
+    hasItalicText,
+    hasBoldText,
+    hasRegularText,
+    hasMediumText,
+    hasNoMargin,
+    hasMargin,
+    hasSectionMargin,
+    hasCenteredText,
+    hasDisabledText,
+    hasCapitalisedText,
+    hasUpperCasedText,
+    intent,
     children,
-    rest,
+    ...rest
+  }: TaggedTypographyProps) =>
+  () => {
+    const intentClass = intent ? getIntentClassName(intent) : null
+
+    return {
+      className: cx(
+        elTextBase,
+        baseClass,
+        hasGreyText && elHasGreyText,
+        hasRegularText && elHasRegularText,
+        hasBoldText && elHasBoldText,
+        hasItalicText && elHasItalicText,
+        hasMediumText && elHasMediumText,
+        hasMargin && elHasMargin,
+        hasNoMargin && elHasNoMargin,
+        hasSectionMargin && elHasSectionMargin,
+        hasCenteredText && elHasCenteredText,
+        hasDisabledText && elHasDisabledText,
+        hasCapitalisedText && elHasCapitalisedText,
+        hasUpperCasedText && elHasUpperCasedText,
+        intentClass,
+        className,
+      ),
+      children,
+      ...rest,
+    }
   }
-}
 
-export const Text3XL: FC<TypographyProps> = (props) => {
-  const { className, children, rest } = useMemo(propsToClass(props), [props])
-  return (
-    <ElText3XL className={className} {...rest}>
-      {children}
-    </ElText3XL>
-  )
+export const TaggedTypography: FC<TaggedTypographyProps> = (props) => {
```

Comment:
Can you add some test cases for this please?

Reply:
should've merged from the branch I deleted ;)